### PR TITLE
cantera-website: Add Gentoo Cantera package installation instructions.

### PR DESCRIPTION
**Pull request by band-a-prend**
This commit adds information about installation of Cantera 2.4.0
package in Gentoo Linux using Portage package system.

New `https://cantera.org/install/index.html` page after patch (note that Ubuntu and Gentoo rows are spanned special to avoid creation of third row but I could move the new instruction to `Other Linux Distributions Install Instructions` page):  

(initial view, see new one in the conversation below)
install

New `https://cantera.org/install/gentoo-install.html` page after patch:  

(initial view, see new one in the conversation below)
gentoo-install

### CODE REVIEW

**Comment by band-a-prend** [comment]
This request is related to Cantera Users' Group topic: 
[Cantera 2.4.0 is now available in Gentoo main portage tree](https://groups.google.com/forum/?hl=en#!topic/cantera-users/Q0AyxpnEGfk)

**Comment by band-a-prend** [comment]
As there no margins between spanned "Ubuntu" and "Gentoo" blocks then should I change `custom.css` rules for `section-card` and `row` classes or is it possible to change their properties locally? I dont't know is it a good practice to add additional class like `section-card-spanned` with margins for it?

File: pages/install/install.rst [hunk 1]
```diff
@@ -86,6 +86,17 @@
 
                Install Cantera on Ubuntu using a PPA.
 
+            .. container:: card-header section-card
```
[the remark this answers is not in the document]
Comment (band-a-prend) [reply pages/install/install.rst]: Thank you for review. I'll make appropriate changes.

p.s. I tried to add container class column (in bootstrap.py) to make new card placed under Ubuntu with space between them but had no success to add required margins for it in css.

[the remark this answers is not in the document]
Comment (bryanwweber) [reply pages/install/install.rst]: That's fine, if Gentoo gets its own card, you should be able to copy the existing code and just change the text + move compiling from source to its own row.

**Comment by decaluwe** [comment]
I kind of like the cards with no space. Since they’re both Linux installations, grouping them together works okay (IMHO).

File: pages/install/install.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (band-a-prend) [reply pages/install/install.rst]: I made the proposed changes and now the pages look like:  

_index.html_
install_third_row  

_gentoo-index.html_
gentoo-install_new

**Comment by bryanwweber** [comment]
@decaluwe what do you think of the changes here? I personally prefer the "own card" look but I can see your point.

@band-a-prend on further reflection, I think it would be better to show how to install the Python interface in these instructions, since that is most likely to be the one that people want to install. Can you add those easily?

**Comment by band-a-prend** [comment]
@bryanwweber this package installs python interface by default (for python-3.6 that is gentoo system default branch of python-3.x). The information about it for gentoo users is mentioned as:

> The Cantera Python module for Python 3 (``python`` USE flag with appropriate ``PYTHON_SINGLE_TARGET``, enabled by default).

The user requires to disable it if he don't want to install this interface. So I placed the links to appropriate documentation articles about USE flags as well below this statement.

Sorry, I currently have no idea how add it easily. Maybe it will be like block about `package.accept_keywords` but there are some fine points that are described via the placed links refering more detailed information.

**Comment by bryanwweber** [comment]
@band-a-prend If the Python interface is installed by default, it should say that somewhere. That wasn't clear to me from how the page is written right now. I guess it does say "enabled by default"... Can you split that first list (the one that starts "The available interfaces are"), maybe something like:

> The following interfaces are installed by default:
> * C++ and the rest of the text about C++
> * CTI Tools (I'm not clear what this is, is this ck2cti? Because that should be distributed as part of the Python package as well, so isn't there a conflict?)
> * Python and the rest of the text about Python
> 
> The following additional interfaces can be added by setting the appropriate `USE` flags:
> * Fortran
>
> You can disable any of the default interfaces by setting the appropriate flags... links go here

In terms of adding more detail on this page, how likely is is that a regular Gentoo user will know about that information? In other words, if an average Gentoo user doesn't require the explanation, we don't have to put it here. However, if the average Gentoo user wouldn't know the first thing about the keywords you said, then it should be explained here. That way, this is a one-stop-shop to install what users want to install.

**Comment by decaluwe** [comment]
I generally prefer his first attempt.  a) a whole new row for one card is aesthetically sub-optimal (to me); b) there is a whole lot of white space in the latest approach; and c) at least on my phone, the “build from source” card is a different width from the other cards. 

I might like the current approach better if the bottom card were the same width as the others and centered. 

But these are all personal preferences, of course.

**Comment by band-a-prend** [comment]
@bryanwweber the information that python interface is installed by default is mentioned at string

> The Cantera Python module for Python 3 (``python`` USE flag with appropriate ``PYTHON_SINGLE_TARGET``, enabled by default).

but I agree that this info must be more notable and I must rework it the way you advise. 

The python interface is default and optional while С++ library always installed (it's compiled anyway and gentoo QA requires to not remove the things that are always compiled). 

The CTI tools are ck2cti, ctml_writer that are installed as python-minimal option of cantera itself. The python USE flag activates python-full installation option (and enables/require cti USE flag to be enabled and then it's a part of python installation of course).

The average gentoo user assumed to be known about USE flags concept because it's base knowledge of gentoo package manager: it used while initial system setup; USE flags per package are always displayed in the list of package and their dependencies before install or downloading. So here I placed just reminder as reference to gentoo handbook and wiki.

As for `accept_keywords` there are mainly
* users that use only stable packages; 
* users that use unstable branch for whole system and packages; 
* users that use stable for system components and unstable packages for some software and libraries.

So 2nd and 3rd group are diffenetly know about `accept_keywords` while 1st group could require some additional explanations like mentioned in instruction.
At least text block about USE and accept_keywords (unmasking) could be move before final install command.

So yes, the average gentoo user doesn't require too detailed explanations.
The gentoo system instalation itself hasn't graphical installator; require manual partitioning and formatting via terminal; manual unpacking and base kernel configuration; manual configuration of base system.

**Comment by bryanwweber** [comment]
@band-a-prend 

> So yes, the average gentoo user doesn't require too detailed explanations.

OK, good to know. I have never used Gentoo before so it is all Greek to me 😄 Please also see the comment from @decaluwe about adjusting the width and centering the last card.

@decaluwe To be more specific, I have 2 problems with the squashed-together cards. The first is, what happens when we add another Linux distro? Does that distro get its own card? It seemed best to me to sort it out now and just have each distro have its own card. At some point, we may end up with too many cards, but that seems like a less likely problem... 🤷‍♂

Second, having the two cards squashed together breaks the consistency of the user interface. All of the other cards have a single headline that you can click to get to the detailed instructions. That one would be the only one with two "headline" links, and it would break the user expectation about the cards that is set up on the first row.

I agree the last card should be centered and the same width as the other ones though 😄 How did you get a build of the website from this PR on your phone?

**Comment by band-a-prend** [comment]
I have problem with centering and width of only card in the 3rd row: the width of card is computed value with right/left margin 15px. So if even I place last card at center then I can't adjust it's width to be equal of cards which are 3 in row. What about leave Ubuntu and Gentoo in one row and Other Distr and CompileFrom Source in other row? This reduce free vertical space for short cards and remove problem of center spacing and width f single-in-row card?

**Comment by band-a-prend** [comment]
Now made some changes required by https://github.com/Cantera/cantera-website/pull/82#issuecomment-510038929 and fix width and centering of lone card in last row (use some workarounds within css file to hide borders of cards and to save card-deck width as well):
cantera_install

gentoo_install

**Comment by band-a-prend** [comment]
@bryanwweber 

If this custom.css patch is suitable? I have to trick with card-deck width to make card be able centered and trick with borders of card to add "empty" cards to make center card have the same width as the card above.

**Comment by bryanwweber** [comment]
@band-a-prend I think I found the right way to do the centering with Bootstrap classes, can you check that everything renders correctly on your end?

**Comment by band-a-prend** [comment]
@bryanwweber 
cards look for me slightly  narrower than the [result](https://user-images.githubusercontent.com/18756734/61129509-3c5cce00-a4bd-11e9-8984-5213f32f90b9.png) of one of my previous comment and they both are narrower than initial variant of appropriate web page.

[New view](https://user-images.githubusercontent.com/18756734/61597236-ba844780-ac16-11e9-8e99-6dc7c82e082d.png) of install page (for me).

**Comment by bryanwweber** [comment]
@band-a-prend Yes, I think that was the result of putting in the `.. container:: col-12` (I'm not at all sure why it should be `12` there, but it seems to work). I don't think its a problem, since they're all still the same width. Can you check the mobile view and make sure that the last card remains the same width and flows into all one column?

FWIW, the reason I added the columns there is because only `.col-*` classes are supposed to be children of `.row`.

**Comment by band-a-prend** [comment]
@bryanwweber 

On my smartphone the [new variant](https://user-images.githubusercontent.com/18756734/61615563-006cfa00-ac6f-11e9-99d2-ea86e9f3bab9.png) also has narrower cards in comparison with [original](https://user-images.githubusercontent.com/18756734/61615637-2e523e80-ac6f-11e9-99f5-ea2955bfbceb.png) one.

**Comment by band-a-prend** [comment]
@bryanwweber 

It seems I have short solution by addition of  `flex-fill` for last `card-deck` and `border-0` for left and right `card` in last row

**Comment by band-a-prend** [comment]
New variant of [desktop](https://user-images.githubusercontent.com/18756734/61629757-7d5b9c00-ac8e-11e9-8146-9bc7eea3d272.png) and [mobile](https://user-images.githubusercontent.com/18756734/61629785-91070280-ac8e-11e9-8389-21264b1db03b.png) views of install page.

Note that mobile view is the same for me as local original version of website 
( they differ from page loaded from web directly).

**Comment by bryanwweber** [comment]
@band-a-prend I'm not at all concerned about the reduced width of the cards, as long as they are all the same width. Is there a reason to be concerned about that?

I don't think your last commit reverting most of my changes is correct, for a few reasons:

1. The "top-level" div should have the class `.container` but you deleted that now
2. Only `.col-*` should be children of `.row`, but you deleted that too (see https://getbootstrap.com/docs/4.3/layout/grid/#how-it-works)
3. I don't want to have blank cards because that seems like a bad hack to get the behavior that is possible without adding them. For instance, for people using screen readers, it is not clear why there are blank cards there. Is there a reason to do it this way instead of the way I suggested?

**Comment by band-a-prend** [comment]
@bryanwweber 

1. Top level (string #21) always previously was just `.. container::`  so I just return it to initial state.
2. As I could see from bootstrap docs the maximum allowed number of columns is 12 and it's seems it's a default value so I think the `col-12` could be omitted here.
3. Likely it looks as my slightly maniacal yearning for saving initial `card-deck` widths. So I agree that empty card is bad hack. I just want to make less considerable changes that affect initial grid on the page.  I rebase my last commit again and now it partially lean on your proposed changes (`col-sm-4`) inside card-deck. The bootstrap docs propose change card sizing also with grid classes https://getbootstrap.com/docs/4.0/components/card/#sizing

and I didnt find if it forbid to use it inside card-deck.

**Comment by bryanwweber** [comment]
@band-a-prend 

1. That was a deliberate change to make the class of that top-level `div` to be `.container`, see https://getbootstrap.com/docs/4.3/layout/overview/
2. The `.col` of some sort is necessary as a direct child of the `.row` according to the Bootstrap documentation as I linked. Just because `.col-12` is the default doesn't mean it should be removed. I see why it should be `.col-12` specifically now, though, and I think that one should be correct.
3. There's no need to be maniacal about keeping the same width as originally, it is a very small change in look of the page and makes the HTML/CSS more semantically correct and fit the Bootstrap documentation. That it worked before was probably an accident/luck. If the commit that I made looks good on your machine (except the small change in card widths), then I think it would be better to revert your last commit (drop it in a rebase, I guess) and merge this with my commit as the last one on the branch.

**Comment by band-a-prend** [comment]
@bryanwweber 

thank you for  the detailed clarifications. Yes, your commit looks good on my machine except only those small change in card width.
Then I shall rebase and delete my second (css) and last commits.

I'm would like to store your commit as separate (atomic) change so I aware to change it's message as afraid it change the initial author name.

**Comment by bryanwweber** [comment]
@band-a-prend Feel free to drop all three formatting change commits, I'll save my commit on a separate branch and cherry-pick it back onto this branch once you update it.

**Comment by bryanwweber** [comment]
Thank you for adding this and for working with us to make all the changes we requested 😄

If you have the time and/or energy, I would really be grateful if you wanted to continue helping maintain the website, since it seems you have some expertise with it now. Please feel free to grab an issue from the Issues page and work on it if you want, I'm not working on anything for the website right now!

**Comment by band-a-prend** [comment]
I rebased it with merging conflicts.

**Comment by bryanwweber** [comment]
I didn't see any conflict markers in the file? And everything seemed like it was working right...

**Comment by band-a-prend** [comment]
I sent previous message before updating the page and saw that pr already merged.

There is was small conflict between my first commit and your commit for install.rst file while deleting my css-commit (I made rebase in tmux env of smartphone). It was unexpected for me too. I rarely use git so it was useful situation for me.

Great thanks for review. I got some new about css and bootstrap :)

**Comment by bryanwweber** [comment]
@band-a-prend Thanks again for doing this 😄 Just want to emphasize again if you want to keep working on the site, feel free to pick up an issue 🎉

**Pull request merged**